### PR TITLE
Linting and Improving project organization

### PR DESCRIPTION
**Pull request by maxired**
Hi,

Here is a PR to fix the linting and organization of the project based on  [senecajs/seneca-level-store/](https://github.com/senecajs/seneca-level-store/)

### CODE REVIEW

**Comment by mcdonnelldean** [comment]
@maxired Not sure why this failed, looked like travis hiccuped. Could you try locall again with latest? Just to see if it is resolved now?

**Comment by maxired** [comment]
@mcdonnelldean , I just rebased on top of master, everything runs fine in local, let's see if Travis woke up ;)

**Comment by mcdonnelldean** [comment]
@maxired Looks like that worked :D Couple of changes
1. Can we move to using standard on it's own. (This is what the org is going to do).
2. Add back the doc folder for the mean time, it powers the annotations, I need to see what @rjrodger wants to do with this

File: package.json [hunk 1]
```diff
@@ -18,24 +24,18 @@
...
-    "Nicolas Herment"
-  ],
-  "license": "MIT",
   "dependencies": {
-    "lodash": "2.4.2"
+    "lodash": "3.10.1"
   },
   "devDependencies": {
-    "docco": "0.6.3",
+    "eslint-config-standard": "^4.4.0",
```
Comment (mcdonnelldean) [inline package.json:32]: We need to add docco back

Comment (mcdonnelldean) [inline package.json:32]: Need to move to just standard

Comment (maxired) [inline package.json:32]: Sorry, I didn't know `standard`, will do

Comment (maxired) [inline package.json:32]: Any idea how to do that using lab ?

Comment (mcdonnelldean) [inline package.json:32]: Oh, no, but I know who does! @geek How do we run standard with Lab or do we need to use ESLint? I know it's the same underneath

**Comment by maxired** [comment]
@mcdonnelldean , you might want to have a look now. 
Maybe you want me to rebase the commits before merging ?

File: .travis.yml [hunk 2]
```diff
@@ -1,5 +1,7 @@
 language: node_js
 node_js:
+  - "4.1"
+  - "4.0"
```
Comment (geek) [inline .travis.yml]: feel free to remove iojs from the list.  You can do `"4"` instead of both 4.0 and 4.1

File: lib/index.js [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+      var ent = args.ent
+      var create = ent.id == null
+      var canon = ent.canon$({
+        object: true
+      })
+      var zone = canon.zone
+      var base = canon.base
+      var name = canon.name
+      if (create) {
+        if (ent.id$ != null) {
```
Comment (geek) [inline lib/index.js]: might as well do explicit equals everywhere: `!==`

Comment (maxired) [inline lib/index.js]: Oh, I haven't touch much there, was thinking to review the code later

File: lib/index.js [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+      var q = args.q
+      var all = q.all$
+      // default false
+      var load = _.isUndefined(q.load$) ? true : q.load$
+      // default true
+      listents(seneca, entmap, qent, q, function (err, list) {
+        if (err) {
+          return cb(err)
+        }
+        list = all ? list : list.length > 0 ? list.slice(0, 1) : []
```
Comment (geek) [inline lib/index.js]: thoughts on

```
list = list || []
list = all ? list : list.slice(0, 1)
```

**Comment by geek** [comment]
The way you are enforcing the standard style is fine.  If you use lab you would need an eslintrc file (https://github.com/senecajs/seneca-level-store/blob/master/.eslintrc) and to update your package to depend on the package used by that eslintrc.  After that, just pass `-L` to lab

**Comment by mcdonnelldean** [comment]
@maxired Super, looks awesome. I think the last thing is to move back to doing the linting the way @geek recommends in the comment above. Totally my fault, sorry. Lets keep em all the same :D

**Comment by maxired** [comment]
@mcdonnelldean , then it should be good ;)

**Comment by mcdonnelldean** [comment]
Awesome! @maxired apologies for the back and forth :D We are working out a base line for all repos so it's a little up in the air right now.

**Comment by maxired** [comment]
@mcdonnelldean , thanks for the feedback. 
No worries regarding back and forth ;)

**Pull request merged**